### PR TITLE
Fill project contract address from scrapers

### PR DESCRIPTION
**Pull request by IvanIvanoff**
#### Summary

[//]: # (#### Related PRs)

[//]: # (#### Additional deploy notes)

[//]: # (#### Screenshots)

### CODE REVIEW

File: lib/sanbase/external_services/project_info.ex [hunk 1]
```diff
@@ -121,13 +123,18 @@ defmodule Sanbase.ExternalServices.ProjectInfo do
...
+
       project
       |> find_or_create_initial_ico()
-      |> Ico.changeset(Map.from_struct(project_info))
+      |> Ico.changeset(project_info_map)
       |> Repo.insert_or_update!()
 
       project
-      |> Project.changeset(Map.from_struct(project_info))
+      |> Project.ContractAddress.add_contract(project_info_map)
```
Comment (tspenov) [inline lib/sanbase/external_services/project_info.ex:134]: Maybe use `Map.split` to get only the needed fields for add_contract ?

File: lib/sanbase/external_services/project_info.ex [hunk 2]
```diff
@@ -44,27 +44,29 @@ defmodule Sanbase.ExternalServices.ProjectInfo do
...
+          ticker: ticker,
+          name: name,
+          token_decimals: token_decimals,
+          total_supply: total_supply
+        } = project
+      ) do
     !website_link or !email or !reddit_link or !twitter_link or !btt_link or !blog_link or
       !github_link or !telegram_link or !slack_link or !facebook_link or !whitepaper_link or
-      !ticker or !name or !main_contract_address or !token_decimals or !total_supply
+      !ticker or !name or !Project.has_contract_address?(project) or !token_decimals or
```
Comment (tspenov) [inline lib/sanbase/external_services/project_info.ex:68]: Aren't token decimals also taken from the new table ?

File: lib/sanbase/external_services/project_info.ex (hunk 1 above)
Comment (tspenov) [inline lib/sanbase/external_services/project_info.ex:134]: Maybe use `Map.split` to get only needed fields ?

File: lib/sanbase/model/project/contract_address.ex [hunk 3]
```diff
@@ -21,6 +21,26 @@ defmodule Sanbase.Model.Project.ContractAddress do
     |> cast(attrs, [:address, :decimals, :label, :description, :project_id])
   end
 
+  def add_contract(
```
Comment (tspenov) [inline lib/sanbase/model/project/contract_address.ex:24]: Can you add spec - especially for the `attrs` param

File: lib/sanbase/external_services/project_info.ex (hunk 1 above)
Comment (IvanIvanoff) [reply lib/sanbase/external_services/project_info.ex:134]: This seems to introduce not needed complexity

File: lib/sanbase/external_services/project_info.ex (hunk 2 above)
Comment (IvanIvanoff) [reply lib/sanbase/external_services/project_info.ex:68]: Let me push some more changes

File: lib/sanbase/external_services/project_info.ex (hunk 1 above)
Comment (IvanIvanoff) [reply lib/sanbase/external_services/project_info.ex:134]: The code extracts only the needed fields and the rest are ignored - this pattern is followed throughout sanbase and we haven't used `split` to explicitly do that.

Comment (tspenov) [reply lib/sanbase/external_services/project_info.ex:134]: Hahah, I wrote it twice cause I thought it was not submitted.
Yeah, my idea is to be easily observable what fields are used - so either use it with a subset and have a typespec for this subset or use pattern matching in the function to get only needed.

**Pull request merged**